lib/Saml2/Auth.php

```diff
@@ -144,12 +144,13 @@ class OneLogin_Saml2_Auth
      * Initializes the SP SAML instance.
      *
      * @param array|object|null $oldSettings Setting data (You can provide a OneLogin_Saml_Settings, the settings object of the Saml folder implementation)
+     * @param bool $spValidationOnly if you only as an SP , you should set it to false if not you should set it to true
      *
      * @throws OneLogin_Saml2_Error
      */
-    public function __construct($oldSettings = null)
+    public function __construct($oldSettings = null , $spValidationOnly = true)
```

Comment:
From what I am seeing when looking at this, defaulting this to `true` will change the default functionality here, as before it would always be `false`. For backwards compatibility it would be best to set it to `false`